lib/rtcpeerconnection/edge.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var EventTarget = require('../util/eventtarget');
+var inherits = require('util').inherits;
+var util = require('../util');
+
+var match = typeof navigator !== 'undefined'
+  ? navigator.userAgent.match(/Edge\/(\d+).(\d+)$/)
+  : null;
+var PeerConnection = require('rtcpeerconnection-shim')(window, match[2]);
+
+function EdgeRTCPeerConnection(configuration) {
+  if (!(this instanceof EdgeRTCPeerConnection)) {
+    return new EdgeRTCPeerConnection(configuration);
+  }
+
+  EventTarget.call(this);
+
+  var peerConnection = new PeerConnection(configuration);
+
+  // NOTE(syerrapragada): These properties are not defined on Prototype.
+  // We need these here until upstream releases a new version.
+  // see: https://github.com/otalk/rtcpeerconnection-shim/commit/755ada6cc062ca3abde12d15de7f6c8928409134
+  Object.defineProperties(this, {
+    _peerConnection: {
+      value: peerConnection
+    },
+    iceConnectionState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceConnectionState;
+      }
+    },
+    iceGatheringState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceGatheringState;
+      }
+    },
+    signalingState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.signalingState;
+      }
+    }
+  });
+
+  util.proxyProperties(PeerConnection.prototype, this, peerConnection);
+}
+
+inherits(EdgeRTCPeerConnection, EventTarget);
+
+// NOTE(syerrapragada): otalk/RTCPeerConnection shim throws InvalidAccessError
+// when sender is not part of getSenders or has been previously removed
+// bug: https://github.com/otalk/rtcpeerconnection-shim/issues/154
+EdgeRTCPeerConnection.prototype.removeTrack = function removeTrack(sender) {
+  if (this.signalingState !== 'closed') {
+    if (!this._peerConnection.getSenders().includes(sender)) {
+      return;
+    }
+  }
+  this._peerConnection.removeTrack(sender);
```

Comment:
@syerrapragada Shouldn't the call to `_peerConnection.removeTrack(...)` be inside the `if (this.signalingState !== 'closed')` block just after the second `if(...)`?

Reply:
I should have merged both if conditions into one. It handles the case where track is already removed from the senders list.